include/swift/Serialization/ScanningLoaders.h

```diff
@@ -60,7 +73,8 @@ class SwiftModuleScanner : public SerializedModuleLoaderBase {
   std::vector<std::string> swiftModuleClangCC1CommandLineArgs;
 
 public:
-  std::optional<ModuleDependencyInfo> dependencies;
+  std::vector<SwiftModuleScannerQueryResult::IncompatibleCandidate> incompatibleCandidates;
```

Comment:
I feel it is weird to have a global state here. The best option I think is to change the APIs to return `SwiftModuleScannerQueryResult` directly. If that is not possible, these two fields need to be private and have an API that does `takeScannerQueryResult()` that construct a query result from move constructing from these two fields and clear the value.

Reply:
It's tricky to avoid global state here _right now_ because a lot of the lookup logic here is threaded through normal module loader API, but I really like the `takeScannerQueryResult` idea for now, I'll give that a shot. 
In the future, I plan to further separate the scanner "loader" from plain module loaders by factoring out the common logic, and then we'll be able to get rid of this being global state.